modules/openshift-developer-cli-openshift-cluster-objects.adoc

```diff
@@ -0,0 +1,87 @@
+
+* Configuration files:
+** `supervisor.conf` is the configuration file necessary for the SupervisorD daemon to start.
+* Scripts:
+** `assemble-and-restart` is an OpenShift S2I concept to build and/or deploy user-source code. The assemble-and-restart script first assembles the user source code inside the application container and then restarts `SupervisorD` for user changes to take effect.
+** `Run` is an OpenShift S2I concept of executing the assembled source code. The run script executes the assembled code created by the `assemble-and-restart` script.
+** `s2i-setup` is a script that creates files and directories which are necessary for the `assemble-and-restart` and run scripts to execute successfully. The script is executed whenever the application container starts.
+
+* Directories:
+** `language-scripts`: Openshift S2I allows custom `assemble` and `run` scripts. A few language specific custom scripts are present in the `language-scripts` directory. The custom scripts provide additional configuration to make {odo-title} debug work.
+
+The `emtpyDir Volume` is mounted at `/opt/odo` mount point for both the init container and the application container.
+
+=== `copy-files-to-volume`
+The `copy-files-to-volume` init container copies files that are in `/opt/app-root` in S2I builder image onto the Persistent Volume. The volume is then mounted at the same location (`/opt/app-root`) in an application container.  
+
+Without the `PersistentVolume` on `/opt/app-root` the data in this directory is lost when `PersistentVolumeClaim` is mounted at the same location.
+
+The `PVC` is mounted at `/mnt` mount point inside the init container.
+
+== Application container
+Application container is the main container inside of which the user-source code executes. 
+
+Application container is mounted with two Volumes:
+* emptyDir Volume mounted at /opt/odo
+* The Persistent Volume mounted at /opt/app-root
+
+The `go-init` is executed as the first process inside the application container. The `go-init` process then starts the `SupervisorD` daemon.
+
+The `SupervisorD` executes and monitores the user assembled source code. If the user process crashes, `SupervisorD` restarts it.
+
+== `PersistentVolume` and `PersistentVolumeClaim`
+`PersistentVolumeClaim` (`PVC`) is a volume type in Kubernetes which provisions a `PersistentVolume`. The life of a `PersistentVolume` is independent of a Pod lifecycle. The data on the `PersistentVolume` persists across Pod restarts.
+
+The `copy-files-to-volume` init container copies necessary files onto the `PersistentVolume`. The main application container utilizes these files at runtime for execution.
+
+The naming convention of the `PersistentVolume` is <component-name>-s2idata.
+
+|===
+Container | `PVC Mounted` at
+| `copy-files-to-volume`
+| `/mnt`
+
+| Application container
+| `/opt/app-root`
+|===
+
+== `emptyDir Volume`
```

Comment:
Is this really appropriate being backticked like this? With "volume" in backticks too? (Also below)